### PR TITLE
fix(local): adjust use of local dev dependenices

### PR DESCRIPTION
**Pull request by deer-wmde**
Currently the API image does not work in the local app env.

- Remove `MattAllan\LaravelCodeStyle` from local_providers
- Remove obsolete local check for jobs
- Remove unused IdeHelperServiceProvider
  - This one may be needed by psalm? (edit: I guess not since CI is happy with it atm)

### CODE REVIEW

File: app/Http/Controllers/WikiController.php [hunk 1]
```diff
@@ -110,12 +110,6 @@ public function create(Request $request): \Illuminate\Http\Response
               'wiki_id' => $wiki->id,
             ]);
 
-            // If we are local, the dev environment wont be able to run these jobs yet, so end this closure early.
-            // TODO maybe send different jobs instead? or do this in the jobs?
-            if (App::environment() === 'local') {
```
Comment (m90) [inline app/Http/Controllers/WikiController.php:115]: Just so I understand: why is this not needed anymore?

File: config/app.php [hunk 2]
```diff
@@ -198,7 +198,6 @@
         MattAllan\LaravelCodeStyle\ServiceProvider::class,
     ],
     'local_providers' => [
-        MattAllan\LaravelCodeStyle\ServiceProvider::class,
```
Comment (m90) [inline config/app.php:201]: Can this now be removed from composer.json as well?

Or do we want to configure it to be "auto-discovered" as described here https://github.com/barryvdh/laravel-ide-helper#installation?

File: app/Http/Controllers/WikiController.php (hunk 1 above)
Comment (deer-wmde) [reply app/Http/Controllers/WikiController.php:115]: I'm not sure when this was needed in the first place, but nowadays the jobs that follow that check work. We just had the app env set to `production` everywhere prior to https://github.com/wmde/wbaas-deploy/commit/42a08af63af0d460d952938dce1520d76aa3c7e3#diff-bb11be128453a5ee0b102c8feaa9890809f7f925ac2dee60b620b1c351fdeb65R23, so the check wasn't even running

File: config/app.php (hunk 2 above)
Comment (deer-wmde) [reply config/app.php:201]: I pushed a commit to remove it - I don't know if this was really used by anyone

**Review by m90 — APPROVED** [review]
:broom:

**Pull request merged**